cartography/client/core/tx.py

```diff
@@ -210,3 +213,49 @@ def load_graph_data(
             DictList=data_batch,
             **kwargs,
         )
+
+
+def ensure_indexes(neo4j_session: neo4j.Session, node_schema: CartographyNodeSchema) -> None:
+    """
+    Creates indexes if they don't exist for the given CartographyNodeSchema object, as well as for all of the
+    relationships defined on its `other_relationships` and `sub_resource_relationship` fields. This operation is
+    idempotent.
+
+    This ensures that every time we need to MATCH on a node to draw a relationship to it, the field used for the MATCH
+    will be indexed, making the operation fast.
+    :param neo4j_session: The neo4j session
+    :param node_schema: The node_schema object to create indexes for.
+    """
+    queries = build_create_index_queries(node_schema)
+
+    for query in queries:
+        if not query.startswith('CREATE INDEX IF NOT EXISTS'):
+            raise ValueError('Query provided to `ensure_indexes()` does not start with "CREATE INDEX IF NOT EXISTS".')
+        neo4j_session.run(query)
+
+
+def load(
+        neo4j_session: neo4j.Session,
+        node_schema: CartographyNodeSchema,
```

Comment:
If this function is going to be the only entrypoint for 'load_graph_data', then I hesitate this to make this a required argument, because not everything will have a proper NodeSchema, but we still like the managed transaction aspect of `load_graph_data` for other modules to begin using, if we can't easily onboard them to NodeSchemas.

Maybe rename it to `load_graph_data_from_node_schema`?

Reply:
They can still call `load_graph_data()` directly, that's a public entrypoint.

Reply:
I think calling it `load()` makes it like we're saying "this is the way that we intend for all of this to eventually look"